pkg/provisioner/ironic/ironic.go

```diff
@@ -1782,6 +1802,12 @@ func (p *ironicProvisioner) hardPowerOff() (result provisioner.Result, err error
 	}
 
 	if ironicNode.PowerState != powerOff {
+		if ironicNode.LastError != "" && !force {
+			p.log.Info("hard power off error", "msg", ironicNode.LastError)
+			result, _ = operationFailed(fmt.Sprintf("hardPowerOff operation failed: %s",
```

Comment:
This ends up being show to the user, so let's not include method names.

Reply:
Done.